doc/tutorials/gpu-passthru.rst

```diff
@@ -0,0 +1,155 @@
+
+Hardware Platform Supported
+***************************
+Currently, ACRN has enabled GVT-d on the following platforms:
+
+* Kaby Lake
+* Whiskey Lake
+* Elkhart Lake
+
+BIOS Setting
+**********************
+
+Kaby Lake Platform
+==================
+* set "IGD Minimum Memory" to "64MB" in "Devices → Video → IGD Minimum Memory".
+
+Whiskey Lake Platform
+=====================
+* set "PM Support"  to "Enabled" in "chipset → System Agent (SA) Configuration → Graphics Configuration → PM support".
+* set "DVMT Pre-Allocated" to "64MB" in "chipset → System Agent (SA) Configuration → Graphics Configuration → DVMT Pre-Allocated"
+
+Elkhart Lake Platform
+=====================
+* set "DMVT Pre-Allocated" to "64MB" in "Intel Advanced Menu → System Agent(SA) Configuration → Graphics Configuration → DMVT Pre-Allocated"
+
+Pass-through GPU to Guest
+**************************
+
+1. Copy ``/usr/share/acrn/samples/nuc/launch_win.sh`` to ``install_win.sh``
+
+::
+
+   cp /usr/share/acrn/samples/nuc/launch_win.sh ~/install_win.sh
+
+2. Modify ``install_win.sh`` script to specify the Windows image you use.
+
+3. Modify ``install_win.sh`` script to enable GVT-d.
+
+Add the following commands before ``acrn-dm -A -m $mem_size -s 0:0,hostbridge \``
+
+::
+
+      gpudevice=`cat /sys/bus/pci/devices/0000:00:02.0/device`
+
+      echo "8086 $gpudevice" > /sys/bus/pci/drivers/pci-stub/new_id
+      echo "0000:00:02.0" > /sys/bus/pci/devices/0000:00:02.0/driver/unbind
+      echo "0000:00:02.0" > /sys/bus/pci/drivers/pci-stub/bind
+
+Replace ``-s 2,pci-gvt -G "$2" \`` with ``-s 2,passthru,0/2/0,gpu \``
+
+4. Run the ``launch_win.sh``.
+
+      .. note:: If you want to pass-through GPU to Clear Linux User VM, the steps are
+         the same as above except your script.
+
+Enable GVT-d GOP driver
+***********************
+
+Issue
+======
+When enabling GVT-d, guest OS couldn’t light up physical screen
+before OS driver load. So guest BIOS and
+grub GUI couldn’t be showed on the physical screen.
+
+Reason
+==========
+Physical display is initialized by GOP driver or VBIOS
+before OS driver load, but guest BIOS doesn’t have them.
+
+Solution
+==========
+Integrate GOP driver binary into OVMF as a Dxe driver,
+then guest OVMF could see GOP driver and run it in
+graphic pass-through environment.
+So physical display could be initialized
+by GOP and used by guest BIOS and guest grub.
+
+Steps
+======
+1. fetch ACRN OVMF
+
+::
+
+      git clone https://github.com/projectacrn/acrn-edk2.git
+
+2. fetch vbt and gop driver
+
+Fetch vbt and gop driver from the board manufacturer according to your CPU model name.
+
+3. git apply the following two patches
+
+* `Use-the-default-vbt-released-with-GOP-driver.patch
+  <../_static/downloads/Use-the-default-vbt-released-with-GOP-driver.patch>`_
+* `Integrate-IntelGopDriver-into-OVMF.patch
+  <../_static/downloads/Integrate-IntelGopDriver-into-OVMF.patch>`_
+
+4. add vbt and gop driver into OVMF
```

Comment:
This step number 4 would best be put right after #2. I.e. 1) fetch the drivers, 2) copy them to the right location

Reply:
> This step number 4 would best be put right after #2. I.e. 1) fetch the drivers, 2) copy them to the right location

ok